ui-tests/src/test/resources/features/integrations/sql-connector.feature

```diff
@@ -203,8 +203,61 @@ Feature: SQL Connector
     Then validate that logs of integration "predicates" contains bodies of items "<result>"
 
     Examples:
-      | predicate             | value | result  |
-      | :#id > id             | 3     | 1,2     |
-      | (id+1) < :#id         | 4     | 1,2     |
-      | id in (1, :#id)       | 4     | 1,4     |
-      | id = floor(:#id)      | 4     | 4       |
+      | predicate        | value | result |
+      | :#id > id        | 3     | 1,2    |
+      | (id+1) < :#id    | 4     | 1,2    |
+      | id in (1, :#id)  | 4     | 1,4    |
+      | id = floor(:#id) | 4     | 4      |
+
+  @reproducer
+  @gh-4466
+  @sql-connector-return-keys
+  Scenario: Return generated keys for INSERT statement
+    # insert single item to DB
+    And inserts into "todo" table
+      | task1 |
+
+    When click on the "Create Integration" button to create a new integration
+    Then check visibility of visual integration editor
+    And check that position of connection to fill is "Start"
+
+    # Start step
+    When select the "PostgresDB" connection
+    And select "Periodic SQL invocation" integration action
+    And fill in invoke query input with "SELECT * FROM todo" value
+    And click on the "Next" button
+    Then check visibility of page "Choose a Finish Connection"
+
+    # Finish step
+    When select the "Log" connection
+    And fill in values
+      | Message Context | false |
+      | Message Body    | true  |
+    Then click on the "Done" button
+
+    Then check visibility of page "Add to Integration"
+
+    # Select postgresDB as middle step and insert new value
+    When add integration step on position "0"
+    And select the "PostgresDB" connection
+    And select "Invoke SQL" integration action
+    And fill in invoke query input with "INSERT INTO todo (task, completed) VALUES (:#task, 0)" value
+    And click on the "Next" button
+
+    # Add Data mapper as middle step
+    Then check visibility of page "Add to Integration"
+    When add integration step on position "0"
+    And select "Data Mapper" integration step
+    Then check visibility of data mapper ui
+    When open data mapper collection mappings
+    And create data mapper mappings
+      | task | task |
+    And click on the "Done" button
+    And publish integration
+    And set integration name "Generated_Keys"
+    And click on the "Publish" button
+
+    # Wait for "Running" state and validate
+    And wait until integration "Generated_Keys" gets into "Running" state
+    Then validate that number of all todos with task "task1" is greater than "1"
+    And validate that logs of integration "Generated_Keys" contains bodies of items "2"
```

Comment:
I would simplify this:

1. Periodic invocation with `INSERT INTO todo (task) VALUES ('whatever')`
2. Log step 

then you can just check that you get `{"id": 1}` in the log. No need to select and reinsert the data.

Also, after discussing with @mcada , we'd like to rename the validation step to 
`validate that logs of integration "<integration>" contain items with IDs "<ids>"` to make it clearer, could you please also include this in you PR?

Reply:
So I've simplified it to Timer -> SQL insert -> Log (since having the SQL insert step as start step doesn't return keys right now, issue [#5493](https://github.com/syndesisio/syndesis/issues/5493) is reporting this).
Also, renamed the verification step as requested.